server/subscriptions_test.go

```diff
@@ -1,41 +1,108 @@
 package main
 
 import (
-	"testing"
-
+	"encoding/json"
+	"github.com/mattermost/mattermost-server/v5/plugin/plugintest"
 	"github.com/stretchr/testify/assert"
+	"testing"
 )
 
-func TestSubscriptionFlagsString(t *testing.T) {
-	tcs := []struct {
-		name  string
-		input SubscriptionFlags
-		want  string
+// pluginWithMockedSubs returns mocked plugin for given subscriptions
+func pluginWithMockedSubs(subscriptions []*Subscription) *Plugin {
+	p := NewPlugin()
+	mockPluginAPI := &plugintest.API{}
+
+	subs := Subscriptions{Repositories: map[string][]*Subscription{}}
+	subs.Repositories[""] = subscriptions
+	jsn, _ := json.Marshal(subs)
+	mockPluginAPI.On("KVGet", SUBSCRIPTIONS_KEY).Return(jsn, nil)
+	p.SetAPI(mockPluginAPI)
+	return p
+}
+
+// wantedSubscriptions returns what should be returned after sorting by repo names
+func wantedSubscriptions(repoNames []string, chanelID string) []*Subscription {
+	var subs []*Subscription
+	for _, st := range repoNames {
+		subs = append(subs, &Subscription{
+			ChannelID:  chanelID,
+			Repository: st,
+		})
+	}
+	return subs
+}
+
+func TestPlugin_GetSubscriptionsByChannel(t *testing.T) {
+
+	type args struct {
+		channelID string
+	}
+	tests := []struct {
+		name    string
+		plugin  *Plugin
+		args    args
+		want    []*Subscription
+		wantErr bool
 	}{
 		{
-			name: "should return correct string when one flag set",
-			input: SubscriptionFlags{
-				ExcludeOrgMembers: true,
-			},
-			want: "--exclude-org-member",
+			name: "basic test",
+			args: args{channelID: "1"},
+			plugin: pluginWithMockedSubs([]*Subscription{
+				{
+					ChannelID:  "1",
+					Repository: "asd",
+				},
+				{
+					ChannelID:  "1",
+					Repository: "123",
+				},
+				{
+					ChannelID:  "1",
+					Repository: "",
+				},
+			}),
+			want:    wantedSubscriptions([]string{"", "123", "asd"}, "1"),
+			wantErr: false,
 		},
 		{
-			name: "should return correct string when one flag unset",
-			input: SubscriptionFlags{
-				ExcludeOrgMembers: false,
-			},
-			want: "",
+			name:    "test empty",
+			args:    args{channelID: "1"},
+			plugin:  pluginWithMockedSubs([]*Subscription{}),
+			want:    wantedSubscriptions([]string{}, "1"),
+			wantErr: false,
 		},
 		{
-			name:  "should return correct string when no flag set",
-			input: SubscriptionFlags{},
-			want:  "",
+			name: "test shuffled",
+			args: args{channelID: "1"},
+			plugin: pluginWithMockedSubs([]*Subscription{
+				{
+					ChannelID:  "1",
+					Repository: "c",
+				},
+				{
+					ChannelID:  "1",
+					Repository: "b",
+				},
+				{
+					ChannelID:  "1",
+					Repository: "ab",
+				},
+				{
+					ChannelID:  "1",
+					Repository: "a",
+				},
+			}),
+			want:    wantedSubscriptions([]string{"a", "ab", "b", "c"}, "1"),
+			wantErr: false,
 		},
 	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.plugin.GetSubscriptionsByChannel(tt.args.channelID)
+
+			assert.Equal(t, tt.wantErr, err != nil, "they should be same")
```

Comment:
nit: Shouldn't the message be something more in the lines of "should return no error"?

Reply:
Yes, I think that would be better

Reply:
Sorry, just realized it will not. Returning or not an error depends on the wantErr variable, so you would have to check that variable to create the error message. I think it would be more readable for the test user if you make the exact message. What about something like:
```
message := "should return no error"
if tt.wantErr {
  message := "should return error"
}
asser.Equal(..., message)
```